### DISCUSSION TITLE
Confusion on zustand persist middleware about hydration

### DISCUSSION THREAD

**Discussion by kecoliva** [pmndrs/zustand#1536] [Q&A]
I'm using zustand's persist middleware like this:

```
export const useSidebar = createStore(
  persist(
    () => ({
      sidebarOpen: false,
    }),
    {
      name: "mcc-sidebar",
    }
  )
)
```

I'm getting hydration errors. But it's said [here](https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md) that hydration errors will only occur when using async storage, but I'm using the default localStorage. Am I misunderstanding something here or this is the intended behavior?

**Comment by dai-shi** [comment]
Can you join #1537 discussion?
